drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,481 @@
+{
+	const struct stmpe811_config *config = dev->config;
+
+	/* Enable FIFO above threshold interrupt */
+	int err = i2c_reg_update_byte_dt(&config->bus, STMPE811_INT_EN_REG,
+					 BIT_MASK(8), STMPE811_GIT_FTH);
+
+	if (err < 0) {
+		return err;
+	}
+
+	/* Enable global interrupts */
+	return i2c_reg_update_byte_dt(&config->bus, STMPE811_INT_CTRL_REG,
+				      BIT_MASK(8), STMPE811_GIT_EN);
+}
+
+static int stmpe811_clear_global_it(const struct device *dev, uint8_t source)
+{
+	const struct stmpe811_config *config = dev->config;
+
+	/* Write 1 to the bits that have to be cleared */
+	return i2c_reg_write_byte_dt(&config->bus, STMPE811_INT_STA_REG, source);
+}
+
+static int stmpe811_process(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+
+	int err = stmpe811_ts_get_state(dev);
+
+	if (err) {
+		return err;
+	}
+
+	if (data->touch_detected) {
+		int x = config->screen_width - (data->touch_x / 4096.0) * config->screen_width;
+		int y = config->screen_height - (data->touch_y / 4096.0) * config->screen_height;
+
+		input_report_abs(dev, INPUT_ABS_X, x, false, K_FOREVER);
+		input_report_abs(dev, INPUT_ABS_Y, y, false, K_FOREVER);
+		input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
+	}
+
+	/* Start polling to detect release */
+	data->poll_release_counter = 0;
+	k_work_schedule(&data->polling_work, K_MSEC(config->polling_period_ms));
+
+	return 0;
+}
+
+static void stmpe811_work_handler(struct k_work *work)
+{
+	struct stmpe811_data *data = CONTAINER_OF(work, struct stmpe811_data, work);
+
+	stmpe811_process(data->dev);
+}
+
+static void stmpe811_poll_release_work_handler(struct k_work *work)
+{
+	struct k_work_delayable *polling_work = CONTAINER_OF(work, struct k_work_delayable, work);
+	struct stmpe811_data *data = CONTAINER_OF(polling_work, struct stmpe811_data, polling_work);
+
+	const struct device *dev = data->dev;
+	const struct stmpe811_config *config = dev->config;
+
+	bool poll_again = true;
+	int err;
+	uint32_t current_cycles;
+	uint32_t cycles_diff;
+	uint32_t wait_period_us;
+
+	k_timepoint_t poll_time_end = sys_timepoint_calc(K_MSEC(config->hold_timeout_ms));
+	uint32_t start_period_cycles = k_cycle_get_32();
+	bool touch = stmpe811_ts_detect_touch(dev);
+
+	/* Clear FIFO - data->poll_period_us has to be high enough
+	 * to fill it up before next
+	 */
+
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG, FIFO_STA_CLEAR);
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG,
+						FIFO_STA_OPERATIONAL);
+
+	if (err < 0) {
+		LOG_ERR("Could not clear FIFO after polling (%d)", err);
+	} else {
+		if (!touch) {
+			data->poll_release_counter++;
+		}
+
+		if (data->poll_release_counter >= config->release_counter_threshold ||
+			sys_timepoint_expired(poll_time_end)) {
+			input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
+			stmpe811_clear_global_it(dev, STMPE811_TS_IT);
+			poll_again = false;
+		}
```

Comment:
hey I don't quite follow how this works to be honest, would not you want to keep fetching and reporting x,y coordinates as long as touch is detected and then trigger the release as soon as the touch is gone? I think that with this logic it would feel very laggy, especially if some UI feature is configured to trigger on release. Also what if you press, release and press again before the counter hits?